components/test-suite/src/loci/tests/testng/Configuration.java

```diff
@@ -240,39 +247,48 @@ public String getTileAlternateMD5() {
     return currentTable.get(TILE_ALTERNATE_MD5);
   }
 
-  public Double getPhysicalSizeX() {
+  public Length getPhysicalSizeX() {
     String physicalSize = currentTable.get(PHYSICAL_SIZE_X);
     if (physicalSize == null) return null;
+    String sizeXUnits = currentTable.get(PHYSICAL_SIZE_X_UNITS);
     try {
-      return new Double(physicalSize);
+      return physicalSize == null ? null : FormatTools.getPhysicalSize(new Double(physicalSize), sizeXUnits);
```

Comment:
Many configuration files include

```
PhysicalSizeX = 0.0
PhysicalSizeY = 0.0
```

which then lead to test errors or type

```
  [testng] [2015-10-26 15:27:02,594] [pool-1-thread-4] 	PhysicalSizeZ: FAILED (Series 0 (expected ome.units.quantity.Length: value[0.0], unit[µm] stored as java.lang.Double, actual null))
```

see https://ci.openmicroscopy.org/job/BIOFORMATS-5.1-merge-repository-subset/83/console

For now we might want to handle `physicalSize == 0d` as well in `Configuration.getPhysicalSize` and have a cleanup later.